module/Motion/StraightRunner.cpp

```diff
@@ -75,3 +75,74 @@ void StraightRunner::run(double targetDistance, int pwm)
   // モータの停止
   controller.stopMotor();
 }
+
+// 指定した色を検出するまで直進する
+void StraightRunner::runToColor(COLOR targetColor, int pwm)
+{
+  // pwm値が0の場合はwarningを出して終了する
+  if(pwm == 0) {
+    logger.logWarning(" The pwm value passed to StraightRunner::run is 0 ");
+    return;
+  } else if(targetColor == COLOR::NONE) {
+    logger.logWarning(" The targetColor value passed to StraightRunner::runToColor is NONE ");
+    return;
+  }
+
+  // 直進前の走行距離
+  int initialRightMotorCount = measurer.getRightCount();
+  int initialLeftMotorCount = measurer.getLeftCount();
+  double initialDistance = Mileage::calculateMileage(initialRightMotorCount, initialLeftMotorCount);
+
+  // 直進中の走行距離
+  int currentRightMotorCount = initialRightMotorCount;
+  int currentLeftMotorCount = initialRightMotorCount;
+  double currentDistance = initialDistance;
+  // ループ回数
+  int count = 0;
+  // 現在のpwm値
+  int currentPwm = 0;
+  int sign = (pwm > 0) ? 1 : -1;
+  int error = 0;                // 左右の回転数の誤差
+  Pid pid(0.5, 0.3, 0.001, 0);  // 左右の回転数を合わせるためのPID
+  int adjustment = 0;           // 左右の誤差の補正値
```

Comment:
`Pwm`を付けたほうが分かりやすいかもしれません。
```suggestion
  int adjustmentPwm = 0;        // 左右の誤差の補正値
```